ui/src/components/retreat/home/PraiseWallContainer.js

```diff
@@ -0,0 +1,282 @@
+import {
+  Button,
+  Flex,
+  Spacer,
+  Text,
+  Box,
+  VStack,
+  Icon,
+  Modal,
+  ModalOverlay,
+  Textarea,
+  useDisclosure,
+  ModalContent,
+  ModalHeader,
+  Alert,
+  AlertIcon,
+  AlertDescription,
+  HStack,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Center,
+} from '@chakra-ui/react';
+import { AddIcon } from '@chakra-ui/icons';
+import { BiNote } from 'react-icons/bi';
+
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { customAxios as axios } from '../../helpers/customAxios';
+import PraiseCard from './PraiseCard';
+
+const PraiseWallContainer = (props) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [praiseList, setPraiseList] = useState([]);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+
+  const cardBgColors = ['#6dced3', '#ffdc82', '#f39371', '#ffdc82', '#6dced3'];
+  const eventCategory = 'CWC 2022';
+
+  useEffect(() => {
+    GetPraise();
+  }, []);
+  const SplitToChunks = (array, parts) => {
+    let result = [];
+
+    for (let i = parts; i > 0; i--) {
+      result.push(array.splice(0, Math.ceil(array.length / i)));
+    }
+    return result;
+  };
+  const InputModal = (props) => {
+    const { user } = props;
+    const PostPraise = async ({ message }) => {
+      try {
+        onClose();
+        const { data } = await axios.post('/api/praises/create', {
+          fullName: 'user',
```

Comment:
this isn't passing the username properly. try this to get the username of the logged in user
```
  const user = useSelector((state) => state.user);
```